ApollosProject/apollos-apps
# feat: Comment List Feature

## DESCRIPTION

### What does this PR do, or why is it needed?

Looks for a `comments` attribute on Content Items and includes a CommentListFeature when the value of that attribute is 'True'.

### What design trade-offs/decisions were made?

Rock administrators will need to modify their Content Channel Types to include a new boolean Item Attribute with a key of 'comments' (all lowercase).

### How do I test this PR?

The dev Rock environment already has the attribute added. You should see the CommentListFeature on each ContentItem in the following query:

```
query {
  contentChannels {
    childContentItemsConnection {
      edges {
        node {
          id
          title
          ... on WeekendContentItem {
            featureFeed {
              features {
                id
                ... on CommentListFeature {
                  id
                }
              }
            }
          }
        }
      }
    }
  }
}
```

You can then toggle the boolean to 'No' on any item to observe the CommentListFeature no longer being returned for that item.

### What automated tests did you write?

Added an assertion and to the snapshots for the existing features test.

## TODO:


## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
